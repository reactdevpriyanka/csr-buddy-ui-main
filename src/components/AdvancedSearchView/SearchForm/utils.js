import { subWeeks, isValid } from 'date-fns';
import * as Yup from 'yup';

export const phoneRegExp = /^(\+?\d{0,4})?\s?-?\s?(\(?\d{3}\)?)\s?-?\s?(\(?\d{3}\)?)\s?-?\s?(\(?\d{4}\)?)?$/;

export const validationSchema = Yup.object({
  orderId: Yup.string().trim().matches(/^\d*$/, 'Must be numeric characters'),
  status: Yup.string().trim(),
  name: Yup.string().trim(),
  logonId: Yup.string().trim(),
  address: Yup.string().trim(),
  city: Yup.string().trim(),
  zip: Yup.string().matches(
    /^\d{5}(?:-\d{4})?$/,
    'Must be 5 or 9 digits with a format of ##### or #####-####',
  ),
  email: Yup.string().email().trim(),
  phone: Yup.string()
    .trim()
    .matches(phoneRegExp, 'Phone number is not valid')
    .min(7, 'Phone number is not valid')
    .max(16, 'Phone number is not valid'),
  partNumber: Yup.string().trim(),
  blocked: Yup.string().trim(),
  blockReason: Yup.string().trim(),
  fulfillmentCenter: Yup.string().trim(),
  account: Yup.string().trim().min(4).max(4),
  payReferenceId: Yup.string().trim(),
  paypalEmail: Yup.string().email().trim(),
  ipAddress: Yup.string().trim(),
  timePlacedFrom: Yup.string()
    .trim()
    .test('duration', 'The start date must be before the end date', function (value) {
      return checkDuration(value, this.parent.timePlacedTo, this);
    })
    .nullable(),
  timePlacedTo: Yup.string().trim().nullable(),
  timeUpdatedFrom: Yup.string()
    .trim()
    .test('duration', 'The start date must be before the end date', function (value) {
      return checkDuration(value, this.parent.timeUpdatedTo, this);
    })
    .nullable(),
  timeUpdatedTo: Yup.string().trim().nullable(),
  subscriptionId: Yup.string().trim(),
});

export const OrderfilterFormWhiteList = [
  'id',
  'orderId',
  'memberId',
  'address',
  'email',
  'name',
  'phone',
  'ipAddress',
  'logonId',
  'status',
  'statuses',
  'excludedStatuses',
  'siteId',
  'paypalEmail',
  'payReferenceIds',
  'account',
  'donationOrgId',
  'subscriptionId',
  'parentOrderId',
  'blockReason',
  'fulfillmentCenter',
  'partNumber',
];

export function isValidDateRange(startDateVal, endDateVal) {
  if (startDateVal && endDateVal) {
    const startDate = new Date(startDateVal);
    const endDate = new Date(endDateVal);
    if (startDate.getTime() > endDate.getTime()) {
      return false;
    }
    const twoWeeksAgo = subWeeks(endDate, 2);
    if (startDate.getTime() < twoWeeksAgo.getTime()) {
      return false;
    }
  }
  return true;
}

const orderFiltersOnlySearchWithThemNotAllowed = new Set(['city', 'zip']);

/**
 * Returns a list of errors on a filter form if there are any
 *
 * @param {object} validationSchema // The schema used to validate the current form
 * @param {object} values // The current values for the form
 * @returns {object} // The list of errors the form currently has.  If none then the form is currently valid or not dirty
 */
export const doFilterFormValidation = async (
  validationSchema,
  filterFormWhiteList = [],
  values,
) => {
  let errors = {};

  if (!values) {
    return errors;
  }

  const validationOptions = {
    abortEarly: false,
  };

  await validationSchema.validate(values, validationOptions).catch((error) => {
    for (let i = 0; i < error.inner.length; i++) {
      const currentError = error.inner[i];
      const currErrorParams = currentError.params;

      if (currErrorParams.value !== '' && currErrorParams.value !== null) {
        errors[currErrorParams.path] = currentError.message;
      }
    }
  });

  if (JSON.stringify(errors) === '{}' && filterFormWhiteList.length > 0) {
    const results = Object.entries(values).filter(
      ([name, value]) => value !== '' && value !== null,
    );

    if (results.length === 1 && !filterFormWhiteList.includes(results[0][0])) {
      errors[results[0][0]] = 'There must be at least 2 items to search on';
    }

    if (results.length === 3 && orderFiltersOnlySearchWithThemNotAllowed.has(results[0][0])) {
      errors[results[0][0]] =
        'at least one of memberId, address, name, phone, statuses, excludedStatuses, siteId, paypalEmail, payReferenceIds, account, donationOrgId, subscriptionId, parentOrderId, blockReasons, fulfillmentCenters, partNumbers, or trackingId must be set';
    }
  }

  return errors;
};

export function checkDuration(startDateStr, endDateStr, ctx) {
  if (startDateStr && endDateStr) {
    const startDate = new Date(startDateStr);
    const endDate = new Date(endDateStr);
    if (!isValid(startDate) || !isValid(endDate)) {
      return false;
    }
    if (startDate.getTime() > endDate.getTime()) {
      return false;
    }
    const twoWeeksAgo = subWeeks(endDate, 2);
    if (startDate.getTime() < twoWeeksAgo.getTime()) {
      return ctx.createError({ message: 'The date range should be less than 2 weeks' });
    }
  }
  return true;
}
