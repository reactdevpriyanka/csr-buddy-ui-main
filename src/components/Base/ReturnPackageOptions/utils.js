export const LABEL_CHOICES = [
  {
    label: '1 Label',
    enabled: true,
    visible: true,
    value: 1,
  },
  {
    label: '2 Labels',
    enabled: true,
    visible: true,
    value: 2,
  },
  {
    label: '3 Labels',
    enabled: true,
    visible: true,
    value: 3,
  },
  {
    label: '4 Labels',
    enabled: true,
    visible: true,
    value: 4,
  },
  {
    label: '5 Labels',
    enabled: true,
    visible: true,
    value: 5,
  },
  {
    label: '6 Labels',
    enabled: true,
    visible: true,
    value: 6,
  },
];

export const BOX_CHOICES = [
  {
    label: '1 Box',
    enabled: true,
    visible: true,
    value: 1,
  },
  {
    label: '2 Boxes',
    enabled: true,
    visible: true,
    value: 2,
  },
  {
    label: '3 Boxes',
    enabled: true,
    visible: true,
    value: 3,
  },
  {
    label: '4 Boxes',
    enabled: true,
    visible: true,
    value: 4,
  },
  {
    label: '5 Boxes',
    enabled: true,
    visible: true,
    value: 5,
  },
  {
    label: '6 Boxes',
    enabled: true,
    visible: true,
    value: 6,
  },
];
