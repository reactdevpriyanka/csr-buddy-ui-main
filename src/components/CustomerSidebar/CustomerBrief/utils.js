export const isWeight = (name) => name === 'WEIGHT_LIMIT';

export const blacklistedTags = new Set(['DISCOUNT_ABUSER', 'RETURNS_ABUSER', 'SUSPECTED_FRAUD']);
