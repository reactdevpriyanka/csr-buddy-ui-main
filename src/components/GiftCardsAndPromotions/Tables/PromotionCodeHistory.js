/* eslint-disable jsx-a11y/anchor-is-valid */
import Link from 'next/link';
import PropTypes from 'prop-types';

const PromotionCodeHistory = ({ promotions, classes, customerId }) => {
  return promotions?.length > 0 ? (
    <table className={classes.table} data-testid="promotion-code-history-table">
      <thead>
        <tr>
          <th>ID</th>
          <th>Name</th>
          <th>Order #</th>
        </tr>
      </thead>
      <tbody>
        {promotions.map((promotion) => (
          <tr key={promotion?.promotionId}>
            <td>{promotion?.promotionId ?? '-'}</td>
            <td>{promotion?.promotionName ?? '-'}</td>
            <td>
                {promotion?.orderId  ? (
                  <Link
                    passHref
                    key={promotion?.orderId}
                    href={`/customers/${customerId}/orders/${promotion?.orderId}`}
                  >
                    <a>{promotion?.orderId}</a>
                  </Link>
                ) : (
                  <div>-</div>
                )}
              </td>
          </tr>
        ))}
      </tbody>
    </table>
  ) : null;
};

PromotionCodeHistory.propTypes = {
  promotions: PropTypes.arrayOf(PropTypes.object),
  classes: PropTypes.object,
  customerId: PropTypes.string,
};

export default PromotionCodeHistory;
