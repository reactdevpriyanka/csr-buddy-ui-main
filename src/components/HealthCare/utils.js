import { CLOSE_BUTTONS } from '../Base/BaseDialog';

export const closeConsentDialog = (response, setConsentGiven, setDialogOpen) => {
  switch (response) {
    case CLOSE_BUTTONS.CANCEL_BUTTON:
      setConsentGiven(false);
      break;
    case true:
      setConsentGiven(true);
      break;
    default:
      break;
  }
  setDialogOpen(false);
};

export const insurablePetsFilter = (pets = []) => {
  return pets?.filter(
    (pet) =>
      !(
        pet?.petInsurancePolicies?.some(
          (policy) => policy?.productType === 'Insurance' && policy?.policyStatus === 'Active',
        ) &&
        pet?.petInsurancePolicies?.some(
          (policy) => policy?.productType === 'Wellness' && policy?.policyStatus === 'Active',
        )
      ) &&
      (pet?.type?.name === 'Dog' || pet?.type?.name === 'Cat'),
  );
};
