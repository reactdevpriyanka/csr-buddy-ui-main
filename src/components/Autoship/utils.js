import { getDynamicString } from '@/utils/string';
import { formatDate, isCardExpired } from '@utils/dates';

export const renderTitle = ({ name, stat, classes, isDetails = false, getLang }) => {
  let defaultStr = `Autoship "${name}" Details`;

  if (isDetails) {
    return getDynamicString({
      key: 'autoshipCardTitle',
      fallback: defaultStr,
      substitutions: [name],
      getLang,
    });
  }

  if (stat === Status.CANCELLED) {
    defaultStr = `Cancelled Autoship "${name}"`;
    return (
      <span className={classes.cancelledHeaderTitle}>
        {getDynamicString({
          key: 'autoshipCardCancelTitle',
          fallback: defaultStr,
          substitutions: [name],
          getLang,
        })}
      </span>
    );
  } else if (stat === Status.UPCOMING) {
    defaultStr = `Upcoming Shipment for "${name}" Autoship`;
    return getDynamicString({
      key: 'autoshipCardUpcomingTitle',
      fallback: defaultStr,
      substitutions: [name],
      getLang,
    });
  }
  // else { // fix this after BE type data is avilable
  //   defaultStr = `"${name}" Autoship created`;
  //   return getDynamicString({
  //     key: 'autoshipCardCreatedTitle',
  //     fallback: defaultStr,
  //     substitutions: [name],
  //     getLang,
  //   });
  // }
  else {
    defaultStr = `Upcoming Shipment for "${name}" Autoship`;
    return getDynamicString({
      key: 'autoshipCardUpcomingTitle',
      fallback: defaultStr,
      substitutions: [name],
      getLang,
    });
  }
};

export const Status = {
  UPCOMING: Symbol.for('UPCOMING'),
  CANCELLED: Symbol.for('CANCELED'),
  CREATED: Symbol.for('CREATED'),
};

export const getStat = (isUpcoming, status) =>
  isUpcoming ? Status.UPCOMING : status === 'CANCELED' ? Status.CANCELLED : Status.CREATED;

export const makeSubtitle = ({
  date,
  stat,
  lastShipmentDate,
  startDate,
  showCardExpired,
  showLastShipmentDeclined,
  frequency,
  subscriptionId,
  getLang,
}) => {
  let subtitle = [];

  if (frequency) {
    subtitle.push({ text: `Frequency: ${frequency}` });
  }

  let defaultStr = `Cancelled on ${formatDate(date, false)}`;
  let shipmentDefaultStr = `Next shipment on ${formatDate(date, false)}`;

  const nextDateStr =
    stat === Status.CANCELLED
      ? getDynamicString({
          key: 'autoshipNextDate',
          fallback: defaultStr,
          substitutions: [`${formatDate(date, false)}`],
          getLang,
        })
      : getDynamicString({
          key: 'shipNextDate',
          fallback: shipmentDefaultStr,
          substitutions: [`${formatDate(date, false)}`],
          getLang,
        });

  subtitle.push({
    text: nextDateStr,
  });

  // eslint-disable-next-line unicorn/no-array-push-push
  subtitle.push({ text: `Last shipment on ${formatDate(lastShipmentDate, false)}` });

  if (showCardExpired) {
    subtitle.push({
      text: 'Upcoming shipment will not ship out due to expired card.',
      isError: true,
    });
  }

  if (showLastShipmentDeclined) {
    subtitle.push({
      text: 'Last shipment was declined due to expired card.',
      isError: true,
    });
  }

  if (subscriptionId) {
    subtitle.push({
      text: `Autoship ID: ${subscriptionId}`,
    });
  }

  return subtitle;
};

export const showCardExpired = (subscriptionData) => {
  return subscriptionData?.paymentMethods?.some(({ type, card }) => {
    // Skip showing "expired card" message if PayPal:
    if (type === 'PAYPAL' || !card?.expirationMonth || !card?.expirationYear) {
      return false;
    }
    const { expirationMonth, expirationYear } = card;
    return isCardExpired(expirationMonth, expirationYear);
  });
};

export const getPromotions = (subscriptionData) => {
  return subscriptionData.promotions.reduce((result, item) => {
    const tmpAdjments = item.adjustments.filter((item) => item.type === 'PROMOTION');

    for (const adjustment of tmpAdjments) {
      const obj = result[adjustment.code];

      if (obj) {
        obj.value += adjustment.amount;
      } else {
        result[adjustment.code] = {
          promotion: adjustment.description,
          value: adjustment.amount,
          code: adjustment.code,
        };
      }
    }

    return result;
  }, {});
};
