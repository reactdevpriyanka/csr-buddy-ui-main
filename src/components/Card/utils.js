import { snakeCaseToTitleCase } from '@/utils/string';

const validReturnStates = new Set([
  'SUBMITTED',
  'VERIFYING',
  'RECEIVE_WAIT',
  'PAY_WAIT',
  'GIFT_CARD_CASH_OUT_WAIT',
  'GIFT_CARD_CLOSE_WAIT',
  'GIFT_CARD_CLOSED',
  'VERIFIED',
  'PROCESSING',
  'TAX_WAIT',
  'PROCESSED',
  'CANCELLING',
  'CANCELING',
  'CANCELLED',
  'CANCELED',
  'IMPORTED_IN_FLIGHT',
  'VET_DIET',
  'FAILED',
]);

export const ITEM_TYPE = {
  DISCONTINUED: Symbol.for('DISCONTINUED'),
  OUT_OF_STOCK: Symbol.for('OUT_OF_STOCK'),
  PERSONALIZED: Symbol.for('PERSONALIZED'),
  FROZEN: Symbol.for('FROZEN'),
  CONNECT_WITH_A_VET: Symbol.for('CONNECT_WITH_A_VET'),
  PHARMACY: Symbol.for('PHARMACY'),
  DROPSHIP: Symbol.for('DROPSHIP'),
  FORCED_BACKORDER: Symbol.for('FORCED_BACK_ORDER'),
  BUNDLED_ITEM: Symbol.for('BUNDLED'),
  VET_DIET: Symbol.for('VET_DIET'),
};

export const ITEM_TYPES_LABELS = {
  [ITEM_TYPE.DISCONTINUED]: 'DISCONTINUED ITEM',
  [ITEM_TYPE.OUT_OF_STOCK]: 'OUT OF STOCK',
  [ITEM_TYPE.PERSONALIZED]: 'PERSONALIZED ITEM',
  [ITEM_TYPE.CONNECT_WITH_A_VET]: 'CONNECT WITH A VET',
  [ITEM_TYPE.PHARMACY]: 'PHARMACY ITEM',
  [ITEM_TYPE.DROPSHIP]: 'DROPSHIPPED ITEM',
  [ITEM_TYPE.FORCED_BACKORDER]: 'FORCED BACKORDER',
  [ITEM_TYPE.BUNDLED_ITEM]: 'BUNDLED ITEM',
  [ITEM_TYPE.VET_DIET]: 'VET DIET ITEM',
};

export const formatItemType = (status) => {
  return ITEM_TYPES_LABELS[Symbol.for(status)] || status;
};

export const notInStock = (tag) => {
  return tag.includes('OUT_OF_STOCK') || tag.includes('FORCED_BACK_ORDER');
};

export const formatReturnType = (enumStr) => {
  switch (enumStr) {
    case 'PRODUCT_CONCESSION':
      return 'Concession';
    default:
      return snakeCaseToTitleCase(enumStr);
  }
};

export const formatReturnState = (state) => {
  switch (state) {
    case 'FAILED':
      return 'Failed';
    case 'CANCELED':
    case 'CANCELLED':
      return 'Canceled';
    case 'PROCESSED':
      return 'Completed';
    default:
      return validReturnStates.has(state?.toUpperCase()) ? 'Pending' : state;
  }
};

export const formatAction = (type, state) =>
  `${formatReturnType(type)} ${formatReturnState(state)}`;
