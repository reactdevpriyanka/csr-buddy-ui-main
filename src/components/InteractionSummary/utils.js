import PhoneCallbackIcon from '@material-ui/icons/PhoneCallback';
import ForumIcon from '@material-ui/icons/Forum';
import EmailIcon from '@material-ui/icons/Email';
import { snakeCaseToTitleCase } from '@/utils/string';
import { useMemo } from 'react';
import _groupBy from 'lodash/groupBy';

export const getContactReason = (details) => {
  if (details?.tertiary) return snakeCaseToTitleCase(details.tertiary);
  else if (details?.secondary) return snakeCaseToTitleCase(details.secondary);
  else if (details?.primary) return snakeCaseToTitleCase(details.primary);
  else return null;
};

export const getContactChannelIcon = (channel) => {
  switch ((channel || '').toLowerCase()) {
    case 'chat':
      return <ForumIcon />;
    case 'email':
    case 'online form':
      return <EmailIcon />;
    default:
      return <PhoneCallbackIcon />;
  }
};

export const useShippingAppeasements = (appeasements) => {
  return useMemo(() => {
    let includeShippingConcession = appeasements.some(
      (appeasement) => appeasement?.description === 'shipping',
    );

    if (!includeShippingConcession) {
      return appeasements;
    }

    let mergedAppeasements = [];
    let shippingAppeasements = appeasements.filter(
      ({ actions }) => actions?.[0]?.type === 'SHIPPING_CONCESSION',
    );
    let shippingAppeasementsGroupedById = _groupBy(
      shippingAppeasements,
      (appeasement) => appeasement.appeasementId,
    );

    // filter out shipping concession appeasements before we push merged shipping appeasements in
    mergedAppeasements = appeasements.filter(
      ({ actions }) => actions?.[0]?.type !== 'SHIPPING_CONCESSION',
    );

    // Merge the shipping concessions (dummy shipping line item is added in ConcessionTable on submit)
    for (const appeasementId in shippingAppeasementsGroupedById) {
      let mergedShippingAppeasement = shippingAppeasementsGroupedById[appeasementId].reduce(
        (merged, currAppeasement) => {
          return { ...currAppeasement, ...merged };
        },
        {},
      );

      mergedAppeasements.push(mergedShippingAppeasement);
    }

    return mergedAppeasements;
  }, [appeasements]);
};
