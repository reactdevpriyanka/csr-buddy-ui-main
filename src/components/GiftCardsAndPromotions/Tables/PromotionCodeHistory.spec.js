import { renderWrap } from '@/utils';
import promotionsHistoryData from '__mock__/promotions/promotionsHistory';
import PromotionCodeHistory from './PromotionCodeHistory';

describe('<PromotionCodeHistory />', () => {
  const render = renderWrap(PromotionCodeHistory, {
    defaultProps: {
      promotions: promotionsHistoryData,
      classes: { table: {} },
    },
  });

  test('it should render the promotions history table', () => {
    expect(render()).toMatchSnapshot();
  });
});
