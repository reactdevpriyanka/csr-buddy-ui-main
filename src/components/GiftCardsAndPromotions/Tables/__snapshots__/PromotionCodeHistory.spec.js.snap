// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromotionCodeHistory /> it should render the promotions history table 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="[object Object]"
        data-testid="promotion-code-history-table"
      >
        <thead>
          <tr>
            <th>
              ID
            </th>
            <th>
              Name
            </th>
            <th>
              Order #
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              7972
            </td>
            <td>
              (2022-06) BLUE BOX Spend 100 Get 25 GC Targeted
            </td>
            <td>
              <a
                href="/customers/undefined/orders/1074580661"
              >
                1074580661
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="[object Object]"
      data-testid="promotion-code-history-table"
    >
      <thead>
        <tr>
          <th>
            ID
          </th>
          <th>
            Name
          </th>
          <th>
            Order #
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            7972
          </td>
          <td>
            (2022-06) BLUE BOX Spend 100 Get 25 GC Targeted
          </td>
          <td>
            <a
              href="/customers/undefined/orders/1074580661"
            >
              1074580661
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
